api/proto/teleport/legacy/types/types.proto

```diff
@@ -993,6 +993,102 @@ message ProvisionTokenSpecV2 {
   ];
 }
 
+// ProvisionTokenV3 specifies provisioning token
+message ProvisionTokenV3 {
+  option (gogoproto.goproto_stringer) = false;
+  option (gogoproto.stringer) = false;
+
+  // Kind is a resource kind
+  string Kind = 1 [(gogoproto.jsontag) = "kind"];
+  // SubKind is an optional resource sub kind, used in some resources
+  string SubKind = 2 [(gogoproto.jsontag) = "sub_kind,omitempty"];
+  // Version is version
+  string Version = 3 [(gogoproto.jsontag) = "version"];
+  // Metadata is resource metadata
+  Metadata Metadata = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "metadata"
+  ];
+  // Spec is a provisioning token V3 spec
+  ProvisionTokenSpecV3 Spec = 5 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "spec"
+  ];
+}
+
+// ProvisionTokenV3List is a list of provisioning tokens.
+message ProvisionTokenV3List {
+  // ProvisionTokens is a list of provisioning tokens.
+  repeated ProvisionTokenV3 ProvisionTokens = 1;
+}
+
+message ProvisionTokenSpecV3 {
+  // Roles is a list of roles associated with the token, that will be converted
+  // to metadata in the SSH and X509 certificates issued to the user of the
+  // token.
+  repeated string Roles = 1 [
+    (gogoproto.jsontag) = "roles",
+    (gogoproto.casttype) = "SystemRole"
+  ];
+  // JoinMethod is the joining method required in order to use this token.
+  // Supported joining methods include:
+  // - "token"
+  // - "ec2"
+  // - "iam"
```

Comment:
If only these values are supported, can you create an enum to make it explicit?

Reply:
What are the issues here? `jsonpb` supports marshaling and unmarshaling to proto enum names instead of values.